components/tree/index.jsx

```diff
@@ -54,16 +54,6 @@ class Tree extends React.Component {
 		});
 	}
 
-	shouldComponentUpdate (nextProps, nextState) {
```

Comment:
It turns out that it's simpler to avoid the react lifecycle complete by moving treeHasFocus out of state.

Reply:
What I'm seeing here is something like `treeHasFocusDueToClick`? Would that be more accurate? I'll go ahead and push that variable name if you agree.

Reply:
Yes that's right, go ahead.  Thanks!